### PR TITLE
Change form pages to unapproved on applicant edit

### PR DESCRIPTION
**Pull request by StevenMaude**
If a form page is approved, and subsequently edited by the applicant, rescind the approved status.

### CODE REVIEW

File: applications/wizard.py [hunk 1]
```diff
@@ -98,11 +98,11 @@ def get_unbound_data_form(self):
     def get_unbound_approval_form(self):
         return self.approval_form_class(instance=self.instance)
 
-    def get_bound_data_form(self, data, initial=None):
+    def get_bound_data_form(self, data, **kwargs):
```
Comment (StevenMaude) [inline applications/wizard.py:101]: Not entirely sure what's better here, possibly `**kwargs`?

File: applications/views.py [hunk 2]
```diff
@@ -127,11 +127,15 @@ def page(request, pk, key):
     except ValueError:
         raise Http404
 
+    unbound_form = page.get_unbound_data_form()
+
     if request.method == "GET":
-        form = page.get_unbound_data_form()
+        form = unbound_form
```
Comment (ghickman) [inline applications/views.py:133]: Not sure we need this optimisation any more!

File: applications/views.py [hunk 3]
```diff
@@ -127,11 +127,15 @@ def page(request, pk, key):
...
     else:
         form = page.get_bound_data_form(request.POST)
-
+        if form.has_changed():
+            page.instance.is_approved = False
+            page.instance.save()
```
Comment (ghickman) [inline applications/views.py:138]: nit: could we keep a blank both before and after this block?  It can help to denote a separate block for some.

**Review by ghickman — CHANGES_REQUESTED** [review]
Couple of tiny nits, nearly there!

File: applications/views.py (hunk 2 above)
Comment (StevenMaude) [reply applications/views.py]: :eyes: I got blinded to what I'd changed from what was originally present; thanks! Just edited the commit.

**Pull request merged**